### ISSUE TITLE
Improvement of the result given by Calamari after training on IAM  database

### ISSUE THREAD

**Issue by Tailor2019** [Calamari-OCR/calamari#189]
Hello,
@ChWick 
After training Calamari on IAM Database (https://fki.tic.heia-fr.ch/databases/iam-handwriting-database), when I use the command of prediction to a test image it returns a ""Avreage Sentece Confidence=30%"" which is a very  low value.
As a solution I'm trying to add new convolutioons layers and filters .
I find in this file "calamari/calamari_ocr/ocr/backends/network_meta.py /" the default parameters of the network:

**"""def default_network_meta():
    return {
        "inter_threads": 0,
        "intra_threads": 0,
        "ctc_merge_repeated": True,
        "use_peepholes": False,
        "dropout": False,
        "solver": "Adam",
        "ctc": "Default",
        "learning_rate": 1e-3,
        "momentum": 0.9,
        "layers": [],
        "cudnn": True,
        "features": 40,                     # i. e. the line height
    }"""**
I don't seen where there is the definition of the convolutions layers of the default model of the network and how I can add new layers or doing some modification.
Please can help me for resolving these problems.
 
Thanks in advance!

**Comment by ChWick** [comment]
**IMPORTANT**: The predict script **DOES NOT** evaluate the CER, thus the performance of the network. The average sentence confidence is a value that corresponds to the "belief" of the network (which is usually higher than its performance)
After calling "predict" you must use the "eval" script to obtain the performance. Note that handwritten text recognition usually requires a more sophisticated preproc pipeline (e.g. for slant correction)

**Comment by Tailor2019** [comment]
Thanks very much for this explanation!
@ChWick 
in my case the database is very big how can I proceed  to do slant correction for all images (automatically) ?
Thanks in advance

**Comment by ChWick** [comment]
Calamari currently offers no preprocessing step for slant correction. You must perform this manually, or train the model without (might result in a higher error rate)

**Comment by Tailor2019** [comment]
Thanks!!
@ChWick 
When I run the script ''"eval.py"" as I understand there some steps:

1-Print confusions: what is the logic of this method and why you use "lamda" in calculation of this parameter?
2-print_worst_lines:What is the criteria of worst  line I don't seen that?
3-create a """xlsx""" file where the system organise all variables 
Is this is correct ?
Thanks a lot for all your explanation.
